src/generator/passes.jl

```diff
@@ -995,6 +998,18 @@ function (x::ProloguePrinter)(dag::ExprDAG, options::Dict)
             println(io, "using CEnum")
             println(io)
         end
+
+        if wrap_variadic_function
+            println(io, "to_c_type(t::Type) = t\n")
+        end
+
+        if enable_to_c_type_default_conversion
+            println(io, """
+            to_c_type(::Type{<:Vector}) = Ptr{eltype(t)}
```

Comment:
It's very tricky to make this method for `Vector` type work in a generic way. For example, when passing a Julia `String` vector to a `char**`, the target type in the `ccall` signature should be either `Ptr{Cstring}` or `Ptr{Ptr{Cuchar}}`. `Ptr{eltype(t)}` will be `Ptr{String}` in this case, which is not correct.